aws/resource_aws_appautoscaling_policy.go

```diff
@@ -430,6 +425,33 @@ func resourceAwsAppautoscalingPolicyImport(d *schema.ResourceData, meta interfac
 	return []*schema.ResourceData{d}, nil
 }
 
+func validateAppautoscalingPolicyImportInput(idParts []string) (string, string, string, string, error) {
+	var serviceNamespace, resourceId, scalableDimension, policyName string
+
+	if len(idParts) < 4 || len(idParts) > 6 {
```

Comment:
just checked the doc (`aws application-autoscaling describe-scaling-policies help`).
* `resourceId` may contain from 0 to 3 "/" characters
* `scalableDimension` seems to be without "/"
* `policyName` can also contain variable number of "/": 0-3

It would be almost better to
* choose different field delimiter
* or, import by `serviceNamespace/policyName` (if it is unique in namespace)
* or, import by ARN

Reply:
Hi @zarnovican

Regarding your suggestions...

### Choose different field delimiter
If you check the [Terraform import documentation](https://www.terraform.io/docs/providers/aws/r/appautoscaling_policy.html#import) you will see the format Terraform chose for the input in all 3 cases (ecs, rds and dynamodb). Acording to that format, I also made 2 unit tests in `TestValidateAppautoscalingPolicyImportInput_positiveCases` and `TestValidateAppautoscalingPolicyImportInput_negativeCases`.

I might be wrong (I'm really new to golang), but as far as I can see the only possible delimiter is `/` (this is the input coming from the import command) and the problem is, as you mention, that it varies depending on the `serviceNamespace`. This is basically the root of the issue this pr tries to resolve.

Another option I found is to leave the code as it is, truncating the last item in the slice `idParts` which is the `tableName` (which also happens to be redundant, because its coming as the third element in the slice). But this will also force us to append this `tableName` element to the policy, and this will look pretty ugly in the code with something like this:

```golang
// truncating last redundant element
if idParts[0] == "dynamodb" {
	idParts = idParts[:len(idParts) -1]
}

serviceNamespace = idParts[0]
resourceId = strings.Join(idParts[1:len(idParts)-2], "/")
scalableDimension = idParts[len(idParts)-2]
policyName =  idParts[len(idParts)-1]

// appending tableName to the policy
if idParts[0] == "dynamodb" {
	policyName = policyName + "/" + idParts[2]
}
```

It removes the need for the switch, but adds the truncating and appending.

### Import by other fields (serviceNamespace/policyName or ARN)
I think this involves design changes that are out of scope for this issue. Seems like a nice suggestion, but I feel is up to the Terraform team to do some analysis on this. Maybe @nywilken can shed some light on this.